File: submissions/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = nums[0];
        int j = 1;
        int max = nums[0];
        while(j < nums.length){
            
            if(sum < 0) sum = nums[j];
            else sum += nums[j];
            max = Math.max(max, sum);
            j++;
        }
        return max;
    }
}

File: submissions/0053-maximum-subarray/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0;
    let max = Number.MIN_SAFE_INTEGER;
    for(let i=0; i<nums.length; i++){
        if(sum < 0){
            sum = nums[i];
            max = Math.max(sum,max);
        }
        else{
            sum += nums[i];
            max = Math.max(sum,max);
        }
    }
    return max;
};
